ganja.js

```diff
@@ -1454,7 +1454,7 @@
           /^\d+[.]{0,1}\d*[E][+-]{0,1}\d*|^\.\d+[E][+-]{0,1}\d*|^0x\d+|^\d+[.]{0,1}\d*|^\.\d+|^\(\/.*[^\\]\/\)/g,                       // 3: literal hex, nonsci numbers and regex (surround regex with extra brackets!)
           /^(\.Normalized|\.Length|\.\.\.|>>>=|===|!==|>>>|<<=|>>=|=>|\|\||[<>\+\-\*%&|^\/!\=]=|\*\*|\+\+|\-\-|<<|>>|\&\&|\^\^|^[{}()\[\];.,<>\+\-\*%|&^!~?:=\/]{1})/g,   // 4: punctuator
           /^[A-Za-z0-9_]*/g]                                                                                                            // 5: identifier
-        while (txt.length) for(t in tokens) if(resi=txt.match(tokens[t])){ tok.push([t|0,resi[0]]); txt=txt.slice(resi[0].length); break;} // tokenise
+        while (txt.length) for(t in tokens) if (resi = txt.match(tokens[t])) { var len = resi[0].length; if (len == 0) { var charCodes = []; for (var i=0; i<txt.length; i++) { var charCode = txt.charCodeAt(i); if (charCode > 255) charCodes.push(charCode); else break; } var codeString = String.fromCharCode(...charCodes); tok.push([t | 0, codeString]); txt = txt.slice(charCodes.length); } else { tok.push([t | 0, resi[0]]); txt = txt.slice(resi[0].length); } break; } // tokenise
```

Comment:
Why 255?

Reply:
Hey @eric-wieser, this way it will include all unicode characters in the identifier and stop whenever there is a normal/non-unicode character like a space, equal sign, comma, semicolon etc.

Reply:
Why is `255` your boundary for "normal" vs "not normal"?

I think the fix here is to ensure the regexes use the `u` unicode flag, rather than hacking something wrong-but-in-more-subtle-ways on the top.